requirements/sso-requirements.in

```diff
@@ -1,2 +1,3 @@
-pyOpenSSL>=22.1.0  # min version for cryptography==39.0.0
+pyOpenSSL>=23.2.0  # for cryptography>=41
```

Comment:
If `cryptography` includes a pre-built SSL, why do we also need to specify the OpenSSL version here?

Reply:
This might be out of scope, but if we're including `pyOpenSSL` simply for the `crypto` feature, as done [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/sso/certificates.py#L3) (the only reference I see in our codebase), I think it's worthwhile to consider [pyOpenSSL's advice](https://pypi.org/project/pyOpenSSL/):
> Note: The Python Cryptographic Authority strongly suggests the use of [pyca/cryptography](https://github.com/pyca/cryptography) where possible. If you are using pyOpenSSL for anything other than making a TLS connection you should move to cryptography and drop your pyOpenSSL dependency.

It doesn't look like we're using the library to make a TLS connection

Reply:
Actually, I'm confused about the relationship between these two libraries. I think I need to learn some more about that. But I'm keeping the status quo for now and just upgrading the libraries we have in our requirements files.